llap-tez/src/test/org/apache/hadoop/hive/llap/tezplugins/TestLlapTaskSchedulerService.java

```diff
@@ -904,6 +904,48 @@ private void forceLocalityTest1(boolean forceLocality) throws IOException, Inter
     }
   }
 
+  @Test(timeout = 10000)
```

Comment:
this unit test passes also without the patch, could you please include some assertion proving this fix? I believe it should assert that in case of no available hosts request.resetLocalityDelayInfo() is not called

Reply:
Sure, TaskInfo can not be mocked directly so had to play with the adjustedLocalityDelay.
This should do though